lib/kafka_producer.js

```diff
@@ -157,10 +244,54 @@ KafkaProducer.prototype.getWholeMsg = function getWholeMsg(topic, message, timeS
 
 KafkaProducer.prototype.close = function close(callback) {
     var self = this;
+    if (self.batching) {
+        self.flushEntireCache(callback);
+    }
     self.enable = false;
     if (self.restClient) {
         self.restClient.close();
     }
 };
 
+// The batch message payload should follow:
+// 4 BE bytes number of messages + 4 BE bytes size of message + actual message
+function MessageBatch(topic) {
+    var self = this;
+
+    self.payload = null;
+    self.timestamp = new Date().getTime();
+    self.numMessages = 0;
+    self.sizeBytes = 0;
+    self.topic = topic;
+}
+
+MessageBatch.prototype.addMessage = function addMessage(message) {
```

Comment:
Thanks, this is great! However, message is either a string or a buffer. Can we do write with a buffer?